### PR TITLE
Fix constant dimension

### PR DESCRIPTION
**Pull request by Ig-dolci**
# Description
Fix constant array dimension error.

### CODE REVIEW

File: firedrake/adjoint_utils/constant.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def _ad_copy(self):
         return self._constant_from_values()
 
     def _ad_dim(self):
-        return numpy.prod(self.dat.data_ro.cdim)
+        return numpy.prod(self.dat.data_ro.ndim)
```
Comment (Ig-dolci) [inline firedrake/adjoint_utils/constant.py:111]: @JDBetteridge  is it `self.dat.data_ro.ndim` here?

Comment (JDBetteridge) [inline firedrake/adjoint_utils/constant.py:111]:
```suggestion
        return self.dat.cdim
```

**Review by JDBetteridge — CHANGES_REQUESTED** [review]
30% sure...

**Review by connorjward — COMMENTED** [review]
Could we get a simple test of this? Even just a test which is "this code should not raise an exception" might be suitable.
Then discussions of "I think this is the right thing to do" can be validated.

**Review by dham — APPROVED** [review]

**Pull request merged**